server/src/routes/articles.js

```diff
@@ -0,0 +1,101 @@
+const express = require("express");
+const Blog = require("../models/Blog");
+require('dotenv').config();
+require("../utils/cloudinary");
+const cloudinary = require('cloudinary');
+
+const router = express.Router();
+
+router.get("/articles", async (req, res) => {
+  const articles = await Blog.find();
+  res.send(articles);
+});
+
+
+router.post("/articles", async (req, res) => {
+  try {
+    cloudinary.v2.uploader.upload(req.files.image.tempFilePath,(error,result) => {
+      const articles = new Blog({
+        title: req.body.title,
+        content: req.body.content,
+        blogImage: result.url,
+        publicId: result.public_id,
+      });
+      articles.save();
+      res.send(articles);
+      
+    });
+    
+  } catch (error) {
+    console.log("This is the error", error);
+    
+  }
+});
+
+
+router.get("/articles/:id", async (req, res) => {
+  try {
+    const articles = await Blog.findOne({ _id: req.params.id });
+    res.send(articles);
+  } catch {
+    res.status(404);
+    res.send({ error: "Article doesn't exist!" });
+  }
+});
+
+router.patch("/articles/:id", async (req, res) => {
+  try {
+    cloudinary.v2.uploader.upload(req.files.image.tempFilePath, async (error,result) => {
+      await Blog.findOneAndUpdate({
+        _id: req.params.id,
+        blogImage: result.url,
+        publicId: result.public_id,
+       });
+    });
+    res.status(200);
+  } catch {
+    res.status(400);
+    res.send({ error: "Image doesn't exist!" });
+  }
+
+  try {
+    const articles = await Blog.findOne({
+      _id: req.params.id,
+      })
+
+    if (req.body.title) {
+      articles.title = req.body.title;
+    }
+
+    if (req.body.content) {
+      articles.content = req.body.content;
+    }
+
+    await articles.save();
+    res.send(articles);
+    
+  } catch {
+    res.status(404);
+    res.send({ error: "Article doesn't exist!" });
+  }
+});
+
+router.delete("/articles/:id/:publicId", async (req, res) => {
+    try {
+      cloudinary.v2.uploader.destroy(req.params.publicId, async (error,result) => {
```

Comment:
You should check if the article id found then u delete if not u return the message that the article not found

Reply:
Added in the catch in case article id is not  found @niyodusengaclement

Reply:
@niyodusengaclement I have added authentication and validation as well message to be displayed. Kindly review and let me know if I can move forward with testing.